packages/react/collection/src/Collection.tsx

```diff
@@ -17,21 +15,39 @@ function createCollection<ItemElement extends HTMLElement, ItemData>() {
   };
   const Context = React.createContext<ContextValue>({} as any);
 
-  const COLLECTION_SLOT_NAME = 'CollectionSlot';
+  const PROVIDER_NAME = 'CollectionProvider';
 
-  const CollectionSlot = React.forwardRef<CollectionElement, SlotProps>((props, forwardedRef) => {
+  const CollectionProvider: React.FC = (props) => {
     const { children } = props;
     const ref = React.useRef<CollectionElement>(null);
-    const composedRefs = useComposedRefs(forwardedRef, ref);
     const itemMap = React.useRef<ContextValue['itemMap']>(new Map()).current;
     return (
       <Context.Provider value={React.useMemo(() => ({ itemMap, collectionRef: ref }), [itemMap])}>
-        <Slot ref={composedRefs}>{children}</Slot>
+        {children}
       </Context.Provider>
     );
-  });
+  };
+
+  CollectionProvider.displayName = PROVIDER_NAME;
+
+  /* -----------------------------------------------------------------------------------------------
+   * CollectionRootSlot
+   * ---------------------------------------------------------------------------------------------*/
+
+  const ROOT_SLOT_NAME = 'CollectionRootSlot';
+
+  type SlotProps = React.ComponentProps<typeof Slot>;
+
+  const CollectionRootSlot = React.forwardRef<CollectionElement, SlotProps>(
```

Comment:
Can we name this `CollectionSlot` still? Curious why we needed the name change.

Reply:
I renamed it because I thought it was now clearer that it was about the DOM slots, one for the `root` and one for each `item`.

Happy to revert that bit though if you think it doesn't help anything.

Reply:
Yeah I don't think it is helping because usually we name our components like `Menu` and `MenuItem` not `MenuRoot` and `MenuItem`. 

I ignore the `Slot` part of the name as I read it mentally like [this](https://github.com/radix-ui/primitives/pull/778#issuecomment-879169603).